nemo/core/classes/modelPT.py

```diff
@@ -112,22 +113,22 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
                 kwargs will be built and supplied to instantiate the optimizer.
         """
         # Setup optimizer and scheduler
-        if 'sched' in optim_config and self._cfg is not None and 'trainer' in self._cfg.pl:
-            if self._cfg.pl.trainer.max_steps is None:
-                if self._cfg.pl.trainer.gpus == 0:
+        if 'sched' in optim_config and self._trainer is not None:
+            if not isinstance(self._trainer.accumulate_grad_batches, int):
+                raise ValueError("We do not currently support gradient acculumation that is not an integer.")
+            if self._trainer.max_steps is None:
+                if self._trainer.gpus == 0:
```

Comment:
Do we need to check if `self._trainer.gpus` is an int, or a list prior to this? Or are we guaranteed an int by this point?